modules/contracts/src.ts/services/ethService.ts

```diff
@@ -106,6 +108,17 @@ export class EthereumChainService extends EthereumChainReader implements IVector
     return signer;
   }
 
+  private async updateNonce(chainId: number, usedNonce: number, signer: Signer): Promise<void> {
```

Comment:
again, only nitpick I have here is that this should only ever be called within the queue, so I don't like this code existing outside of it.

if someone called from elsewhere, it would be disaster. Obvi no one will, but still... idk

Reply:
Fixed